packages/databricks-vscode/src/configuration/BundleConfigReaderWriter.ts

```diff
@@ -168,19 +170,22 @@ export class BundleConfigReaderWriter {
     ) {
         const file = await this.getFileToWrite(key, target);
         if (file === undefined) {
-            return false;
+            throw new Error(
+                `Can't find a file to write property '${key}' of target '${target}'.`
+            );
         }
         const data = await parseBundleYaml(file);
         const targetData = data.targets?.[target];
         if (targetData === undefined) {
-            return false;
+            throw new Error(`No target '${target}' for writing '${key}.`);
         }
 
         const newTargetData = this.writerMapping[key](targetData, value);
+        if (JSON.stringify(newTargetData) === JSON.stringify(targetData)) {
```

Comment:
Can we move to using deep equality util here? (and in other places)
At least in the past stringify comparison was sensitive to the order of the fields in object (at the creation time).

Reply:
That is a great point. I will make a separate PR to replace all instances of json.stringify. I was thinking of using https://www.npmjs.com/package/deep-equal. Do you have any other library in mind or should we implement something on our own?

Reply:
`lodash.isequal` might be better. We only have to care about primitive values, objects, and arrays, and can technically just create our own utility and don't pull any other additional dependencies, but I think we already have lodash in the bundle, since it's a transitive dependency, might as well make it a direct dependency